### PR TITLE
Bugfix : ignore works now even if different forces or /sth , /nth or /s used

### PR DESCRIPTION
**Pull request by Ragnarok77-factorio**
### Brief description of the changes:
Now ignore should work even when the player is in another force, or use /sth or /nth /s commands.

Only edge case is that admins cant be ignored by default, so if admin and player are in the same force, and player ignores admin, it won't work if same force, but nothing we can do about it as i'ts a factorio "feature" that admins cant be ignored
Second thing, the player text is still shown above the player head, but nothing we can do about it, it's a factorio feature to show text above player name.

Hard to say if we should prevent to ignore admins or not, but I know some players like to ignore some admins so, why not give them the possibility. But they can't complain then if they ignored an admin and get punished later because of not listening admin because they ignored the admin...

Fixes : https://github.com/Factorio-Biter-Battles/Factorio-Biter-Battles/issues/316

### Tested Changes:
- [x] I've tested the changes locally or with people.
- [ ] I've not tested the changes.

### CODE REVIEW

File: maps/biter_battles_v2/functions.lua [hunk 1]
```diff
@@ -351,8 +361,8 @@ function Public.share_chat(event)
 		Muted.print_muted_message(player)
 	end 
 	if not muted and player_force_name == "spectator" then
-		game.forces.north.print(msg)
-		game.forces.south.print(msg)
+		Public.print_message_to_players(game.forces.north.players,player_name,msg,color)
```
Comment (Ragnarok77-factorio) [inline maps/biter_battles_v2/functions.lua:364]: Revert this change, should not have added color

File: maps/biter_battles_v2/functions.lua [hunk 2]
```diff
@@ -318,6 +318,16 @@ function Public.no_landfill_by_untrusted_user(event)
 	end
 end
 
+function Public.print_message_to_players(forcePlayerList,playerNameSendingMessage,msgToPrint,colorChosen)
```
Comment (mysticamber) [inline maps/biter_battles_v2/functions.lua:321]: Blame Cliff for making me look. 

I'm not approving/rejecting, just a comment. 

It's best to establish a standard, camelCase or snake_case, and have some spaces between param lists,etc. There's probably a lua formatter plugin :) It makes it easier to read and review.

File: maps/biter_battles_v2/main.lua [hunk 3]
```diff
@@ -63,6 +63,39 @@ local function on_console_chat(event)
 	Functions.share_chat(event)
 end
 
+local function on_console_command(event)
```
Comment (mysticamber) [inline maps/biter_battles_v2/main.lua:66]: One of the things I dreamt of, back when I was still working on bb code, is unifying "on_x" functions. 👍

File: maps/biter_battles_v2/functions.lua (hunk 1 above)
Comment (mysticamber) [reply maps/biter_battles_v2/functions.lua:364]: colour is nice though

Comment (Ragnarok77-factorio) [reply maps/biter_battles_v2/functions.lua]: fixed now

File: maps/biter_battles_v2/functions.lua (hunk 2 above)
Comment (Ragnarok77-factorio) [reply maps/biter_battles_v2/functions.lua]: Yes I think we will need to discuss about a formatter and standard (camel vs snake) between all devs, and I would like also to basically format all the code, but that would probably happen (if approved) once there is no PR anymore (or barely any.. right now too many, it would become a mess I think if we did that now)

Comment (Ragnarok77-factorio) [reply maps/biter_battles_v2/functions.lua]: added space though now for the params list at this location

File: maps/biter_battles_v2/main.lua (hunk 3 above)
Comment (Ragnarok77-factorio) [reply maps/biter_battles_v2/main.lua:66]: You mean have all on_x of the scenario in the same file location...?

File: maps/biter_battles_v2/functions.lua [hunk 4]
```diff
@@ -318,6 +318,20 @@ function Public.no_landfill_by_untrusted_user(event)
 	end
 end
 
+function Public.print_message_to_players(forcePlayerList, playerNameSendingMessage, msgToPrint, colorChosen)
+	for _, playerOfForce in pairs(forcePlayerList) do
+		if playerOfForce.connected then
+			if global.ignore_lists[playerOfForce.name] == nil or (global.ignore_lists[playerOfForce.name] and not global.ignore_lists[playerOfForce.name][playerNameSendingMessage]) then
+				if color == nil then
```
Comment (developer-8) [inline maps/biter_battles_v2/functions.lua:325]: `color`  is not defined

Comment (Ragnarok77-factorio) [reply maps/biter_battles_v2/functions.lua]: Good call, thanks a lot ! fixed it

**Comment by Ragnarok77-factorio** [comment]
Beware, the case of [PR405](https://github.com/Factorio-Biter-Battles/Factorio-Biter-Battles/pull/405) must be handled if 405 is merged first.

File: maps/biter_battles_v2/main.lua [hunk 5]
```diff
@@ -63,6 +63,39 @@ local function on_console_chat(event)
...
+                global.ignore_lists[player.name][param] = nil
+                player.print("You have unignored " .. param, {r = 0, g = 1, b = 1})
+            else
+                player.print("You are not currently ignoring " .. param, {r = 0, g = 1, b = 1})
+            end
+		else
+			player.print("Invalid input. Make sure the name contains no spaces, quotes, semicolons, backticks, or any spaces.", {r = 1, g = 0, b = 0})
+		end
+	end
+end
```
Comment (zw7u9tru) [inline maps/biter_battles_v2/main.lua:97]: logic seems fine, but difficult to read since you're mixing tabs and spaces here

**Comment by Ragnarok77-factorio** [comment]
Added comment for the regex to explain it.
